fault/magma_simulator_target.py

```diff
@@ -33,13 +47,17 @@ def run(self):
                     value = value.as_uint()
                 simulator.set_value(action.port, value)
             elif isinstance(action, actions.Print):
-                got = BitVector(simulator.get_value(action.port))
+                got = simulator.get_value(action.port)
+                if isinstance(action.port, (m.ArrayType, m.ArrayKind)) and \
+                        isinstance(action.port.T, (m._BitType, m._BitKind)):
+                    got = BitVector(got).as_uint()
+                elif isinstance(action.port, (m.ArrayType, m.ArrayKind)):
```

Comment:
The issue here for printing complex nested arrays is that it doesn't map to our `format` string parameter. Copied my comment below that seemed to get lost in this change set, maybe from the rebase.

> I think we should defer this feature for now. The user can print elements of the array fine, but there's not a simple way to map a complex nested array to a standard format string.

Reply:
Yeah, I think I would like to implement a more general thing, like `Print(format_string, *args)` where we have special syntax for single ports, arrays, and tuples. Think it can be deferred for future work. This is good for now.